mlx_audio/tts/models/outetts/dac_interface.py

```diff
@@ -0,0 +1,158 @@
+import math
+
+import mlx.core as mx
+import numpy as np
+import pyloudnorm as pyln
+import scipy.signal
+import soundfile as sf
+
+from mlx_audio.codec import DAC
+
+
+def process_audio_array(
+    audio: mx.array,
+    sample_rate: int = 24000,
+    target_loudness: float = -18.0,
+    peak_limit: float = -1,
+    block_size: float = 0.400,
+) -> mx.array:
+    audio_np = np.array(audio)
+
+    # handle multi-channel audio
+    if len(audio_np.shape) > 1:
+        if audio_np.shape[1] > 1:
+            audio_np = np.mean(audio_np, axis=1)
+        else:
+            audio_np = np.squeeze(audio_np)
+
+    original_length = len(audio_np)
+    min_samples = int(block_size * sample_rate)
+
+    if original_length < min_samples:
+        pad_length = min_samples - original_length
+        audio_padded = np.pad(audio_np, (0, pad_length), mode="constant")
+    else:
+        audio_padded = audio_np
+
+    # measure and normalize loudness
+    meter = pyln.Meter(sample_rate, block_size=block_size)
+    measured_loudness = meter.integrated_loudness(audio_padded)
+    normalized = pyln.normalize.loudness(
+        audio_padded, measured_loudness, target_loudness
+    )
+
```

Comment:
Btw this library is pretty small and actively maintained.

We could consider having a copy here as a module.